tests/code_update.py

```diff
@@ -4,153 +4,210 @@
 import infra.network
```

Comment:
Not too many changes in this file, mainly adding `test_verify_quotes` and wrapping the existing test in a new `test_update_all_nodes`.